Add Linux (gcc) support (#109)
cg-tuwien/Gears-Vk: README.md

* Execute `git submodule update --init --recursive` in order to pull both, _Gears-Vk_ and the [_Auto-Vk_](https://github.com/cg-tuwien/Auto-Vk) framework.
* The steps described under section [Creating a New Project](#creating-a-new-project) might be helpful for setting up a custom Visual Studio project that links agains _Gears-Vk_.

## CMake

> _Gears-Vk_ also supports building with CMake on Linux (*gcc* or *clang*) and Windows (*MSVC*).

Author: fixed in https://github.com/cg-tuwien/Gears-Vk/pull/109/commits/fecb7511821f7ff4bd1a1733d4b350681ab7fa73

It has the following signature:

```Cmake
add_post_build_commands(
        target              # the executable target (e.g. hello_world)
        glslDirectory       # the absolute path of the directory containing GLSL shaders used by the target
        spvDirectory        # the absolute path of the directory where compiled SPIR-V shaders should be written to
        assetsDirectory     # the absolute path of the directory where assets should be copied to (or where symbolic links should be created) - can be a generator expression
        assets              # a list containing absolute paths of assets which should be copied to ${assetsDirectory} - can be files or directories
        symlinks)           # a boolean setting if symbolic links of assets (and DLLs on Windows) should be created instead of copying dependencies
```

All shader files found in the given `glslDirectory` are added to a custom target (named `${target}_shaders`) which is added to the given `target` as a dependency.
Shaders belonging to this custom target will be compiled before the dependent `target` is build.
Whenever a shader file changes they will be recompiled automatically.

> Note, however, that new shader files are only added to the custom target at configure time, so if you're working with an IDE you must make sure to trigger CMake to reconfigure its build files whenever you add a new shader.

Reviewer: ```suggestion
Note, however, that new shader files are only added to the custom target at configure time. This means that if you're working with an IDE, it must be ensured to trigger CMake to reconfigure its build files whenever you add new shader files.
```

I'll give you some of these "you"s ;)

> E.g. on CLion or Visual Studio you can temporarily add a `message()` call as a work-around.

Reviewer: ```suggestion
As a work-around, a `message()` call can be added temporarily in CLion or Visual Studio.
```

Note that creating symbolic links might require the user running CMake to have special privileges. E.g. on Windows the user needs the `Create symbolic links` privilege.
If the user doesn't have the required privileges `add_post_build_commands` falls back to copying the dependencies.

### Creating a New Project (CMake)

> To create a new *Gears-Vk* project using CMake you can use the [Gears-Vk-Starter template](https://github.com/JolifantoBambla/Gears-Vk-Starter).

Reviewer: ```suggestion
A new *Gears-Vk* project based on CMake can conveniently be created on GitHub with the [Gears-Vk-Starter template](https://github.com/JolifantoBambla/Gears-Vk-Starter).
```
